packages/cli/src/cmds/agentInstaller/pythonAgentInstaller.ts

```diff
@@ -146,26 +150,68 @@ export class PipenvInstaller extends PythonInstaller {
     return undefined;
   }
 }
+
 export class PipInstaller extends PythonInstaller {
   static identifier = 'pip';
+  private _buildFile: string = 'requirements.txt';
 
   constructor(path: string) {
     super(PipInstaller.identifier, path);
   }
 
   get buildFile(): string {
-    return 'requirements.txt';
+    return this._buildFile;
   }
 
   get buildFilePath(): string {
     return join(this.path, this.buildFile);
   }
 
   async available(): Promise<boolean> {
-    return exists(this.buildFilePath);
+    return !!(await isFile(this.buildFilePath));
+  }
+
+  async _findBuildFiles(): Promise<string[]> {
+    const choices = await promisify(glob)('*requirements*.txt', {
+      matchBase: true,
+      cwd: this.path,
+      nodir: true,
+    });
+
+    return choices;
+  }
+
+  async _chooseBuildFile(choices): Promise<string> {
+    const defaultChoice = choices.filter((f) => minimatch(f, '*dev*', { matchBase: true }));
+
+    const { buildFile } = await UI.prompt([
+      {
+        type: 'list',
+        name: 'buildFile',
+        message: 'Please choose the requirements file used for development:',
+        choices,
+        default: defaultChoice[0],
+      },
+    ]);
+
+    return basename(buildFile);
   }
```

Comment:
Should these be marked as `private`? We don't typically use a leading underscore to imply visibility, especially when the method is actually public.

If it's a matter of accessibility within tests, something like one of the following would work:
```ts
(installer as unknown) as { _chooseBuildFile(): Promise<string> }
(installer as unknown) as { _findBuildFiles(): Promise<Array<string>> }
```

```ts
interface PythonInstallerInternal {
  _chooseBuildFile() => Promise<string>;
  _findBuildFiles() => Promise<Array<string>>;
}

...

installer as unknown as PythonInstallerInternal
```

Depending on the TS config you may even be able to get away with simply
```ts
installer['_findBuildFiles']
```

Reply:
Yeah, they should be `private`.  The leading underscore was spillover from python. ;)